### PR TITLE
Use common TLS parsing routine for etcd

### PR DESCRIPTION
**Pull request by johnbelamaric**
Change to use the new common routine, and update the docs to reflect
the different options for passing TLS configuration.

### CODE REVIEW

File: middleware/etcd/setup.go [hunk 1]
```diff
@@ -12,6 +10,7 @@ import (
 	"github.com/miekg/coredns/middleware"
 	"github.com/miekg/coredns/middleware/pkg/dnsutil"
 	"github.com/miekg/coredns/middleware/pkg/singleflight"
+	mwtls "github.com/miekg/coredns/middleware/tls"
```
Comment (miekg) [inline middleware/etcd/setup.go]: Just saw. Think we should move this under `middleware/pkg/tls`

**Comment by miekg** [comment]
LGTM

File: middleware/etcd/setup.go (hunk 1 above)
Comment (johnbelamaric) [reply middleware/etcd/setup.go]: So how are the tests passing. WTF?

Comment (johnbelamaric) [reply middleware/etcd/setup.go]: Oh, *should move*. I thought you were saying it *is already* under pkg. I will move it.

**Pull request merged**